include/fastscapelib/profile_grid.hpp

```diff
@@ -207,19 +217,43 @@ namespace fastscapelib
      */
     template <class XT, class C>
     profile_grid_xt<XT, C>::profile_grid_xt(size_type size,
-                                        spacing_type spacing,
-                                        const boundary_status_type& status_at_bounds,
-                                        const std::vector<node>& status_at_nodes)
+                                            spacing_type spacing,
+                                            const boundary_status_type& status_at_bounds,
+                                            const std::vector<node>& status_at_nodes)
         : base_type(size), m_size(size), m_spacing(spacing), m_status_at_bounds(status_at_bounds)
     {
         m_shape = {static_cast<typename shape_type::value_type>(m_size)};
+        m_length = static_cast<spacing_type>(size) * spacing;
```

Comment:
```suggestion
        m_length = static_cast<spacing_type>(size - 1) * spacing;
```

Reply:
looks like a typo... I will modify tests to be able to capture this! thanks!